lib/consoletest.pm

```diff
@@ -48,13 +49,21 @@ Method executed when run() finishes and the module has result => 'fail'
 
 =cut
 sub post_fail_hook {
-    my ($self) = shift;
+    my ($self) = @_;
     $self->SUPER::post_fail_hook;
     select_console('log-console');
     $self->remount_tmp_if_ro;
     $self->export_logs_basic;
     # Export extra log after failure for further check gdm issue 1127317, also poo#45236 used for tracking action on Openqa
     $self->export_logs_desktop;
+
+    # Destroy the public cloud instance in case of fatal test failure
+    # TODO: currently there is theoretical chance to call cleanup two times. See details in poo#95780
```

Comment:
Don't add disabled code or TODO comments. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit or just the ticket you mentioned

Reply:
We can remove `TODO` if it confuse you . this comment have two goals : 
1. actual "TODO" action item
2. hint for a person who will debug the case when cleanup will be called to times and we will get into troubles

But I personally don't see any problem with having `TODO` statements in the code , especially when it is **addition** to the ticket so it won't be lost. But while ticket is in the scope of some certain team this comment might motivate someone outside this team to pick it up ( I understand that chances for that are lower than meet dinosaur but don't take away our dream :D ) . 
Also can you actually elaborate **why** it is bad to add TODO comments ? ( no doubts about disabled code )

Reply:
TODO comments will rot. Especially considering how little people care in this project about the overall quality comments like "TODO" or "FIXME" will be ignored by others or just confuse them, not knowing if these things really need to be done or not. People read and change the code because they want to achieve something. I am sure that no one will look for all "TODO" references in code and say "oh yeah, I have time, I can now think hard what the original author from years ago meant here and do it". Also see https://blog.usejournal.com/stop-writing-code-comments-28fef5272752

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12972

Reply:
> People read and change the code because they want to achieve something.

Exactly . And this is my expectation . If `cleanup` will be called 2 times someone will reach this line see the comment and will fix it because it ate his/her time 

> I am sure that no one will look for all "TODO" references in code and say "oh yeah, I have time, I can now think hard what the original author from years ago meant here and do it"

and this is not the expectation :) see above . Also you don't need to "think hard" because this TODO has ticket 

> https://blog.usejournal.com/stop-writing-code-comments-28fef5272752

the problem of this article that it showing cases how to write comments in a bad way but I am not sure if this proves that there is no good ways 

I wonder if just "TODO" word will be removed without change any text this will solve problem for you ?

Reply:
> > People read and change the code because they want to achieve something.
> 
> Exactly . And this is my expectation . If `cleanup` will be called 2 times someone will reach this line see the comment and will fix it because it ate his/her time

yeah right, show me when that happened ;)

> I wonder if just "TODO" word will be removed without change any text this will solve problem for you ?

hm, yes